Noalias on a view on a reshape view

Using noalias on a view on a reshape_view produces a compilation error:

```cpp
xtensor<int, 1> a {0};
noalias(view(reshape_view(a, {1}), 0)) = 1;
```

fails with error:

```cpp
xtensor/xassign.hpp:246:57: error: no matching function for call to ‘xt::xview<xt::xstrided_view<xt::xtensor_container<xt::uvector<int, xsimd::aligned_allocator<int, 16> >, 1, (xt::layout_type)1, xt::xtensor_expression_tag>&, std::array<long unsigned int, 1>, (xt::layout_type)0, xt::detail::flat_adaptor_getter<xt::xtensor_container<xt::uvector<int, xsimd::aligned_allocator<int, 16> >, 1, (xt::layout_type)1, xt::xtensor_expression_tag>&, (xt::layout_type)1> >, int>::strides() const’
                    (e1.layout() != layout_type::dynamic && e2.has_linear_assign(e1.strides()));
                    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

```
Lib versions:
xtl_target_version="0.6.4"
xtensor_target_version="0.20.7"
xsimd_target_version="7.2.3"

This was working with xtensor 0.19.4

Note that the following combinations are still compiling:

```cpp
xtensor<int, 1> a {0};
noalias(view(a, 0)) = 1;
noalias(reshape_view(a, {1})) = 1;
noalias(reshape_view(view(a, 0), {1})) = 1;
```

--- a/include/xtensor/xassign.hpp
+++ b/include/xtensor/xassign.hpp
@@ -240,10 +240,16 @@ namespace xt
         }
 
         template <class E1, class E2>
-        inline bool is_linear_assign(const E1& e1, const E2& e2)
+        inline auto is_linear_assign(const E1& e1, const E2& e2) -> std::enable_if_t<has_strides<E1>::value, bool>
         {
             return (E1::contiguous_layout && E2::contiguous_layout && linear_static_layout<E1, E2>()) ||
                    (e1.layout() != layout_type::dynamic && e2.has_linear_assign(e1.strides()));
+        }
+
+        template <class E1, class E2>
+        inline auto is_linear_assign(const E1&, const E2&) -> std::enable_if_t<!has_strides<E1>::value, bool>
+        {
+            return false;
         }
 
         template <class E, class = void>

--- a/test/test_xview.cpp
+++ b/test/test_xview.cpp
@@ -1464,4 +1464,14 @@ namespace xt
         int res = v.element(idx.cbegin(), idx.cend());
         EXPECT_EQ(res, 2);
     }
+
+    TEST(xview, view_reshape_view)
+    {
+        xtensor<int, 1> a = { 0, 1, 2 };
+        xtensor<int, 1> b = { 2, 3, 4 };
+        auto tmp = reshape_view(a, {3});
+        auto res = view(std::move(tmp), xt::all());
+        noalias(view(reshape_view(a, {3}), xt::all())) = b;
+        EXPECT_EQ(tmp, res);
+    }
 }

